controllers/storagecluster/storageconsumer.go

```diff
@@ -321,18 +374,27 @@ func getLocalVolumeGroupSnapshotClassNames(ctx context.Context, kubeClient clien
 		}); err != nil {
 			return nil, err
 		}
+		destToSrcMap := util.GetDestToSrcClassMapping(classesSpec)
 		for idx := range volumeGroupSnapshotClassesInCluster.Items {
-			// TODO: skip volumegroupsnapshotclasses that are from external mode if both internal & external mode is enabled
 			vgsc := &volumeGroupSnapshotClassesInCluster.Items[idx]
-			if slices.Contains(supportedCsiDrivers, vgsc.Driver) {
+			_, aliasOrRename := destToSrcMap[vgsc.Name]
+			if slices.Contains(supportedCsiDrivers, vgsc.Driver) && !aliasOrRename {
```

Comment:
This change is wrong. a rename only applies to the destination not to the source! You don't select based on renames